etc/cmake/options.cmake

```diff
@@ -35,6 +35,11 @@ if(OT_BACKBONE_ROUTER)
     target_compile_definitions(ot-config INTERFACE "OPENTHREAD_CONFIG_BACKBONE_ROUTER_ENABLE=1")
 endif()
 
+option(OT_BIG_ENDIAN "enable big endian byte order")
```

Comment:
Cleanup in this file should rather target the upstream OpenThread repository, we don't want the OT codebase to get out of sync with the upstream (it'll make future upmerges more difficult).

If we're tight on schedule we can merge this here first, but the upstream PR shall be opened nonetheless. But please separate this change in a separate commit, as it will need to be reverted once the upstream gets this change.